src/components/ebay-menu-button/component.js

```diff
@@ -11,26 +10,22 @@ module.exports = {
         }
     },
 
-    toggleItemChecked(itemEl) {
-        const itemIndex = indexOf(itemEl.parentNode.children, itemEl);
-        const item = this.state.items[itemIndex];
-        const currentIndex = this.state.items.findIndex(radioItem => radioItem.checked);
+    toggleItemChecked(index, itemEl) {
+        const currentIndex = this.state.checkedItems.findIndex(checked => checked);
 
-        if (this.state.type === 'radio' && itemIndex !== currentIndex) {
-            this.state.items.forEach((eachItem, i) => {
-                eachItem.checked = i === itemIndex;
-            });
-
-            this.setStateDirty('items');
+        if (this.type === 'radio' && index !== currentIndex) {
+            this.setState('checkedItems', this.state.checkedItems.map((item, i) => i === index));
             this.emitComponentEvent({
+                index,
                 eventType: 'change',
                 el: itemEl
             });
-        } else if (this.state.type !== 'radio') {
-            item.checked = !item.checked;
-            this.setStateDirty('items');
+        } else if (this.type !== 'radio') {
+            this.state.checkedItems[index] = !this.state.checkedItems[index];
```

Comment:
Thoughts on renaming the state variable to `isChecked`? `checkedItems` seems a little off to me because it is not an array of checked items, it is an array of booleans which map to if an item is checked.

Reply:
Done but for type as `radio`, not `type !== radio`